app/uk/gov/hmrc/securemessage/connectors/ChannelPreferencesConnector.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.securemessage.connectors
+
+import play.api.{ Configuration, Logging }
+import play.api.http.Status.{ BAD_GATEWAY, OK }
+import play.api.libs.json.{ JsError, JsResult, JsString, JsSuccess, JsValue, Json, Reads }
+import uk.gov.hmrc.emailaddress.EmailAddress
+import uk.gov.hmrc.http.{ HeaderCarrier, HttpClient, HttpResponse }
+import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import uk.gov.hmrc.securemessage.models.core.Identifier
+
+import javax.inject.{ Inject, Singleton }
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Failure, Success, Try }
+
+@Singleton
+@SuppressWarnings(Array("org.wartremover.warts.All"))
```

Comment:
just wondering, do we really need to suppress all wartremover warnings or was that just left over?